system/autoware_default_adapi/src/planning.cpp

```diff
@@ -49,16 +78,121 @@ std::vector<typename rclcpp::Subscription<T>::SharedPtr> init_factors(
 }
 
 template <class T>
-T merge_factors(const rclcpp::Time stamp, const std::vector<typename T::ConstSharedPtr> & factors)
+std::vector<T> convert(const std::vector<PlanningFactor> & factors)
+{
+  static_assert(sizeof(T) == 0, "Only specializations of convert can be used.");
+  throw std::logic_error("Only specializations of convert can be used.");
+}
+
+template <>
+std::vector<VelocityFactor> convert(const std::vector<PlanningFactor> & factors)
+{
+  std::vector<VelocityFactor> velocity_factors;
+
+  for (const auto & factor : factors) {
+    if (factor.behavior != PlanningFactor::SLOW_DOWN && factor.behavior != PlanningFactor::STOP) {
+      continue;
+    }
+
+    if (factor.control_points.empty()) {
+      continue;
+    }
+
+    if (conversion_map.count(factor.module) == 0) {
+      continue;
+    }
+
+    VelocityFactor velocity_factor;
+    velocity_factor.behavior = conversion_map.at(factor.module);
+    velocity_factor.pose = factor.control_points.front().pose;
+    velocity_factor.distance = factor.control_points.front().distance;
+
+    velocity_factors.push_back(velocity_factor);
+  }
+
+  return velocity_factors;
+}
+
+template <>
+std::vector<SteeringFactor> convert(const std::vector<PlanningFactor> & factors)
 {
-  T message;
+  std::vector<SteeringFactor> steering_factors;
+
+  for (const auto & factor : factors) {
+    if (
+      factor.behavior != PlanningFactor::SHIFT_RIGHT &&
+      factor.behavior != PlanningFactor::SHIFT_LEFT &&
+      factor.behavior != PlanningFactor::TURN_RIGHT &&
+      factor.behavior != PlanningFactor::TURN_LEFT) {
+      continue;
+    }
+
+    if (factor.control_points.size() < 2) {
+      continue;
+    }
+
+    if (conversion_map.count(factor.module) == 0) {
+      continue;
+    }
+
+    if (direction_map.count(factor.behavior) == 0) {
+      continue;
+    }
+
+    SteeringFactor steering_factor;
+    steering_factor.behavior = conversion_map.at(factor.module);
+    steering_factor.direction = direction_map.at(factor.behavior);
+    steering_factor.pose = std::array<geometry_msgs::msg::Pose, 2>{
+      factor.control_points.front().pose, factor.control_points.back().pose};
+    steering_factor.distance = std::array<float, 2>{
+      factor.control_points.front().distance, factor.control_points.back().distance};
+
+    steering_factors.push_back(steering_factor);
+  }
+
+  return steering_factors;
+}
+
+template <class T>
+T merge_factors(
+  const rclcpp::Time stamp, const std::vector<PlanningFactorArray::ConstSharedPtr> & factors)
+{
+  static_assert(sizeof(T) == 0, "Only specializations of merge_factors can be used.");
+  throw std::logic_error("Only specializations of merge_factors can be used.");
+}
```

Comment:
clang-tidy is throwing the following error:
> /__w/autoware.universe/autoware.universe/system/autoware_default_adapi/src/planning.cpp:158:22: error: unused parameter 'stamp' [clang-diagnostic-unused-parameter]
  const rclcpp::Time stamp, const std::vector<PlanningFactorArray::ConstSharedPtr> & factors)
                     ^
/__w/autoware.universe/autoware.universe/system/autoware_default_adapi/src/planning.cpp:158:86: error: unused parameter 'factors' [clang-diagnostic-unused-parameter]
  const rclcpp::Time stamp, const std::vector<PlanningFactorArray::ConstSharedPtr> & factors)
                                                                                     ^
```suggestion
T merge_factors(
  [[maybe_unused]] const rclcpp::Time stamp, [[maybe_unused]] const std::vector<PlanningFactorArray::ConstSharedPtr> & factors)
{
  static_assert(sizeof(T) == 0, "Only specializations of merge_factors can be used.");
  throw std::logic_error("Only specializations of merge_factors can be used.");
}
```